3542/1/lib/repo.rb

```diff
@@ -0,0 +1,63 @@
+require 'httparty'
+require 'nokogiri'
+require 'open-uri'
+
+#:reek:TooManyInstanceVariables
+class Repo
+  URI = 'https://api.github.com/search/repositories'.freeze
+
+  attr_reader :name, :used_by, :watched_by, :stars, :forks, :contributors, :issues
+
+  # rubocop:disable Metrics/AbcSize
+  # rubocop:disable Security/Open
+  def initialize(gem_name)
+    @api_response = HTTParty.get(URI, query: { q: gem_name })
+    @repo = repository
+    @html = Nokogiri::HTML(open(@repo[:html_url]))
+    @name = @repo[:name]
+    @stars = @repo[:watchers_count]
+    @forks = @repo[:forks_count]
+    @issues = @repo[:open_issues_count]
+    @contributors = contributors_count
+    @watched_by = watched_by_count
+    @used_by = used_by_count
+  end
+  # rubocop:enable Metrics/AbcSize
+  # rubocop:enable Security/Open
+
+  private
+
+  def repository
+    symbolize @api_response.to_hash['items'].first
+  end
+
+  # rubocop:disable Style/RescueModifier
+  def symbolize(hash)
```

Comment:
is it really necessary to symbolize this hash?

Reply:
> is it really necessary to symbolize this hash?

it is unnecessary, but write hash keys in symbols more convenient for me. OK, i will remove this method.